activesupport/lib/active_support/backtrace_cleaner.rb

```diff
@@ -104,27 +104,33 @@ def add_stdlib_silencer
         add_silencer { |line| line.start_with?(RbConfig::CONFIG["rubylibdir"]) }
       end
 
-      def filter_backtrace(backtrace)
-        @filters.each do |f|
-          backtrace = backtrace.map { |line| f.call(line) }
+      # Returns the original collection if the block returns an empty collection.
+      def empty_guard(collection)
+        result = yield
+        result.empty? ? collection.dup : result
+      end
+
+      # process +ary+ via +filters+ using +method+, ensuring
+      # _something_ gets returned.
+      def process_collection(ary, filters, method)
+        empty_guard ary do
+          filters.reduce(ary) { |bt, f| bt.send(method) { |line| f.call(line) } }
         end
+      end
 
-        backtrace
+      # use @filters to transform the backtrace via map
+      def filter_backtrace(backtrace)
+        process_collection backtrace, @filters, :map
       end
 
+      # use @silencers to reject parts of the backtrace
       def silence(backtrace)
-        @silencers.each do |s|
-          backtrace = backtrace.reject { |line| s.call(line) }
-        end
-
-        backtrace
+        process_collection backtrace, @silencers, :reject
       end
 
       def noise(backtrace)
-        backtrace.select do |line|
-          @silencers.any? do |s|
-            s.call(line)
-          end
+        empty_guard backtrace do
+          backtrace.select { |line| @silencers.any? { |s| s.call(line) } }
```

Comment:
The strictness is the question... It IS strict on minitest, but this looks to be used in multiple locations so it doesn't necessarily apply here. It might be better to have a subclass JUST for `Minitest.backtrace_filter`. I can go either way.

Reply:
P.S. I lived the first 5ish years of my life in/around Lafayette!

Reply:
@zenspider Haha, small world!  :earth_americas:  :smile: